org.scala-ide.sdt.core/src/org/scalaide/ui/internal/editor/decorators/bynameparams/CallByNameParamAtCreationPresenter.scala

```diff
@@ -0,0 +1,103 @@
+
+  import CallByNameParamAtCreationPresenter._
+
+  protected override def findAll(compiler: ScalaPresentationCompiler, scu: ScalaCompilationUnit, sourceFile: SourceFile): Map[Annotation, Position] =
+    CallByNameParamAtCreationPresenter.findByNameParamCreations(compiler, scu, sourceFile, prefStoreCfg)
+}
+
+object CallByNameParamAtCreationPresenter extends HasLogger {
+  final case class Cfg(firstLineOnly: Boolean)
+
+  private def prefStoreCfg: Cfg = {
+    val prefStore = IScalaPlugin().getPreferenceStore
+    Cfg(prefStore.getBoolean(CallByNameParamCreationPreferencePage.PFirstLineOnly))
+  }
+
+  /**
+   * Finds all places in the source where call-by-name parameters are created.
+   *
+   * See #1002340 for further information.
+   */
+  def findByNameParamCreations(compiler: IScalaPresentationCompiler, scu: ScalaCompilationUnit, sourceFile: SourceFile, cfg: Cfg): Map[Annotation, Position] = {
+    def findByNameParamCreations(tree: compiler.Tree) = {
+
+      object traverser extends compiler.Traverser {
+        var result: Map[Annotation, Position] = Map()
+
+        override def traverse(tree: compiler.Tree) {
+          result = tree match {
+            case compiler.Apply(fun, args) if (fun.tpe != null) => result ++ processArgs(fun.tpe, args)
+            case _ => result
+          }
+
+          super.traverse(tree)
+        }
+
+        def processArgs(funTpe: compiler.Type, args: List[compiler.Tree]): Map[Annotation, Position] = {
+          if (funTpe.params.size != args.size) {
+            // This might happen for code that does not compile cleanly; run the Unit-Tests for this class with an appropriate breakpoint
+            // if you are interested in details.
+            Map()
+          } else {
+            val byNameArgs = funTpe.params.zip(args).withFilter { case (param, _) =>
+               param.isByNameParam
+            }.map(_._2)
+
+            (for (arg <- byNameArgs) yield {
+              val txt = toText(arg)
+              (toAnnotation(arg, txt), toPosition(arg, txt))
+            }).toMap
+          }
+        }
+
+        def toText(arg: compiler.Tree): String = {
+          sourceFile.content.view(arg.pos.start, arg.pos.end).mkString("")
+        }
+
+        def toAnnotation(arg: compiler.Tree, txt: String): Annotation = {
+          new CallByNameParamAtCreationAnnotation(s"Call-by-name parameter creation: () => $txt")
+        }
+
+        def toPosition(arg: compiler.Tree, txt: String): Position = {
+          val start = arg.pos.start
+          val length = {
+            if (cfg.firstLineOnly) {
+              val eol = txt.indexOf('\n')
+              if (eol > -1) eol else txt.length
+            } else {
+              txt.length
+            }
+          }
+          new Position(start, length)
+        }
+      }
+
+      traverser.traverse(tree)
+      traverser.result
+    }
+
+    compiler.askLoadedTyped(sourceFile, false).get match {
```

Comment:
Hm, sorry, I misread the definition of `findAll`. I think it's a mistake that it doesn't pass the tree, since it obtains it in `BaseSemanticAction`, hence forcing all implementers to re-post the request. Hopefully the tree didn't change in the meantime, so it hits the presentation compiler cache.